src/main/java/clueless/Game.java

```diff
@@ -383,15 +384,38 @@ public void loseGame(Player loser) {
 	 * @return
 	 */
 	public Player whoHasClue(List<Card> suggestion) {
-		Player pl;
+		Player pl = null;
 		boolean playerHasClue = false;
-
+		String getPlayerChar = this.getCurrentPlayer().characterName;
+		int currIndex = getCharacterIndexInTurnOrder(getPlayerChar);		
+		int nextPlayerindex = currIndex + 1;
 		int i = 0;
-		//		while ((!playerHasClue) && i < this.characterMap.size()) {
-		//			
-		//		}
-
-		return null;
+		
+			if (nextPlayerindex >= CHARACTER_TURN_ORDER.length) 
+				{ // repoint index if currIndex is last character
+					nextPlayerindex = 0;
+				}
+			
+			while ((!playerHasClue) && i < this.characterMap.size())
```

Comment:
you reference i here but never increment `i`, did you mean to use `count` here? If so, you should initialize count outside this while loop